src/core/mac/sub_mac.cpp

```diff
@@ -356,8 +356,8 @@ void SubMac::StartCsmaBackoff(void)
 
         if (ShouldHandleTransmitTargetTime())
         {
-            if (otPlatRadioGetNow(&GetInstance()) <
-                mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime + mTransmitFrame.mInfo.mTxInfo.mTxDelay)
+            if (Time(static_cast<uint32_t>(otPlatRadioGetNow(&GetInstance()))) <
+                Time(mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime + mTransmitFrame.mInfo.mTxInfo.mTxDelay))
```

Comment:
```suggestion
                Time(mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime) + Time(mTransmitFrame.mInfo.mTxInfo.mTxDelay))
```

Reply:
Note that the `+` operator of `Time` is:
```
Time operator+(uint32_t aDuration) const { return Time(mValue + aDuration); } 
```
There isn't a version that accepts a `Time` type as parameter.

We can change it to:
```
Time(mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime) + mTransmitFrame.mInfo.mTxInfo.mTxDelay
```